### PR TITLE
TGP-2158: Check if user wants to use existing Ukims number

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

File: test/controllers/UseExistingUkimsNumberControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+      }
+    }
+  }
+
+  "onSubmit" - {
+
+    "must return OK and the correct view for a POST" in {
```
Comment (ChrisCookOC) [inline test/controllers/UseExistingUkimsNumberControllerSpec.scala]: Title slightly incorrect :) Should be do the navigating not correct view

File: test/controllers/UseExistingUkimsNumberControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+          routes.UseExistingUkimsNumberController.onSubmit(),
+          ukimsNumber
+        )(
+          request,
+          messages(application)
+        ).toString
+      }
+    }
+  }
+}
```
Comment (ChrisCookOC) [inline test/controllers/UseExistingUkimsNumberControllerSpec.scala:180]: Possible extra test case - if Session Repo fails

File: app/forms/UseExistingUkimsFormProvider.scala [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+package forms
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class UseExistingUkimsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Boolean] =
```
Comment (ChrisCookOC) [inline app/forms/UseExistingUkimsFormProvider.scala:26]: Form provider not tested

File: app/navigation/Navigator.scala [hunk 4]
```diff
@@ -32,7 +32,8 @@ import javax.inject.{Inject, Singleton}
 @Singleton
 class Navigator @Inject() (categorisationService: CategorisationService) {
   private val normalRoutes: Page => UserAnswers => Call = {
-    case ProfileSetupPage                          => _ => routes.UkimsNumberController.onPageLoadCreate(NormalMode)
+    case ProfileSetupPage                          => navigateFromProfileSetUp
```
Comment (ChrisCookOC) [inline app/navigation/Navigator.scala:36]: Can we test the new navigator cases as well please

File: app/utils/Constants.scala [hunk 5]
```diff
@@ -27,4 +27,6 @@ case object Constants {
   val NiphlsCode: String      = "WFE012"
 
   val minimumLengthOfCommodityCode: Int = 6
+
+  val NotFoundErrorCode: Int = 7
```
Comment (ChrisCookOC) [inline app/utils/Constants.scala]: Is this used?

**Review by ChrisCookOC — COMMENTED** [review]
Looks good, noted some missing test cases

File: app/pages/UseExistingUkimsPage.scala [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object UseExistingUkimsPage extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ toString
+
+  override def toString: String = "useExistingUkimsPage"
```
Comment (yohan-oc) [inline app/pages/UseExistingUkimsPage.scala:25]: Shouldn't  we add this to CreateProfileJourney? Otherwise this data will not removed from MongoDb

Comment (ChrisCookOC) [reply app/pages/UseExistingUkimsPage.scala:25]: Sorted

**Review by yohan-oc — APPROVED** [review]

**Pull request merged**